extension.test.js

```diff
@@ -15,22 +32,52 @@ const mockClearDiagnostics = jest.fn()
 const mockDeleteDiagnostics = jest.fn()
 const mockDisposeDiagnostics = jest.fn()
 
+/**
+ * Tag for use with template literals
+ * 
+ * Finds the indentation on the first line after the opening backtick
+ * and removes that indentation from every line in the template.
+ * @param {String[]} strings Array of lines in the template literal
+ */
+function outdent(strings) {
+  // Add in all of the expressions
+  let outdented = strings.map((s, i) => `${s}${arguments[i+1]||''}`).join('')
+  // Find the indentation after the first newline
+  const matches = /^\s+/.exec(outdented.split('\n')[1])
+  if (matches) {
+    const outdentRegex = new RegExp('\\n' + matches[0], 'g')
+    outdented = outdented.replace(outdentRegex, '\n')
+  }
+  return outdented
+}
```

Comment:
@TheSench could you explain why you wanted to "outdent" test strings?

Reply:
This was to prevent changes like adding `describe` blocks from completely changing the test results.  Template literal strings include all of the indentation.  If we ever indent the code they're in, we'd either need to change the snapshots, or have the template literals not be indented with the rest of the code.  I added a comment to one of the affected tests to illustrate this.

Reply:
The `outdent` tag make all of the strings it's tagged with act like they don't have any leading indentation (based on the leading indentation of the second line (the first one starts immediately after the back-tick, so it has none).

Reply:
Here is an example:

By default, this code (`.` to represent whitespace)
```
..mockDocument.text = `
..zero width space
..zero width non-joiner
..non breaking space
..left double quotation mark
..right double quotation mark
..`
```
will produce a string like this:
```

..zero width space
..zero width non-joiner
..non breaking space
..left double quotation mark
..right double quotation mark
..
```

The `outdent` tag will make this
```
..mockDocument.text = outdent`
..zero width space
..zero width non-joiner
..non breaking space
..left double quotation mark
..right double quotation mark
..`
```
behave like this:
```
zero width space
zero width non-joiner
non breaking space
left double quotation mark
right double quotation mark

```

Reply:
ok, I understand the need